src/main/scala/org/broadinstitute/dsde/firecloud/webservice/UserApiService.scala

```diff
@@ -83,43 +83,43 @@ trait UserApiService extends HttpService with PerRequestCreator with FireCloudRe
-                  // rawls error? boo. All we can do is respond with an error.
-                  case InternalServerError => respondWithErrorReport(InternalServerError, InternalServerError.defaultMessage, requestContext)
-                  // rawls found the user; we'll try to parse the response and inspect it
+                  // Sam rejected our request. User is either invalid or their token timed out; this is truly unauthorized
+                  case Unauthorized => respondWithErrorReport(Unauthorized, "Request rejected by identity service - invalid user or expired token.", requestContext)
+                  // Sam 404 means the user is not registered with FireCloud
+                  case NotFound => respondWithErrorReport(NotFound, "FireCloud user registration not found.", requestContext)
+                  // Sam error? boo. All we can do is respond with an error.
+                  case InternalServerError => respondWithErrorReport(InternalServerError, "Identity service encountered an unknown error, please try again. (Error 10)", requestContext)
+                  // Sam found the user; we'll try to parse the response and inspect it
                   case OK =>
                     val respJson = response.entity.as[RegistrationInfo]
                     respJson match {
                       case Right(regInfo) =>
                         if (regInfo.enabled.google && regInfo.enabled.ldap && regInfo.enabled.allUsersGroup) {
-                          // rawls says the user is fully registered and activated!
+                          // Sam says the user is fully registered and activated!
                           requestContext.complete(OK, regInfo)
                         } else {
-                          // rawls knows about the user, but the user isn't activated
-                          respondWithErrorReport(Forbidden, "FireCloud user not activated", requestContext)
+                          // Sam knows about the user, but the user isn't activated
+                          respondWithErrorReport(Forbidden, "FireCloud user not activated.", requestContext)
                         }
-                      // we couldn't parse the rawls response. Respond with an error.
+                      // we couldn't parse the Sam response. Respond with an error.
                       case Left(error) =>
-                        respondWithErrorReport(InternalServerError, InternalServerError.defaultMessage, requestContext)
+                        respondWithErrorReport(InternalServerError, "Identity service encountered an unknown error, please try again. (Error 20)", requestContext)
                     }
                   case x =>
-                    // if we get any other error from rawls, pass that error on
+                    // if we get any other error from Sam, pass that error on
                     respondWithErrorReport(x.intValue, "Unexpected response validating registration: " + x.toString, requestContext)
```

Comment:
might want to tweak this one too, so we pass the full error object up to the UI, instead of just the `toString` of it